### ISSUE TITLE
[feature] Add support for inline breakpoints with column like vscode

### ISSUE THREAD

**Issue by gayanper** [eclipse-platform/eclipse.platform.debug#54]
Today in eclipse we only have line breakpoints. But with more and more LSP/DAP based languages such as JS/TS are supported by Eclipse, it would be nice to have inline breakpoints like in vscode. As the platform it should only be supported to keep the column and then present a codemining with the debug icon at the column in addition to the breakpoints icon in the ruler. may be multiple inline breakpoints on the same line could be shown overlapped on top of each other in the ruler. Clicking on the code mining icon should remove the breakpoint and clicking on the ruler will remove the last breakpoint on the line.

Individual Debug implementation can make use of this feature to implement supported inline breakpoints.

**Comment by gayanper** [comment]
@vogella WDYT ?

**Comment by jonahgraham** [comment]
IMHO this is a great idea. The Platform breakpoints already support this in the data model `ILineBreakpoint.getCharStart()` using the standard `IMarker.CHAR_START` so a lot of the non-ui work in platform may already be done. I don't know if any of the common debuggers already fill in that information, but I know some of the CDT extenders do (or did), but I am not sure how they displayed such information.

**Comment by gayanper** [comment]
Yes i have seen that in the breakpoint model, But i thought it was used only to highlight the debug line. But if this is a good attribute we could introduce codemining as part of platform debug to render them as well. 

But if we use the existing CHAR_START it could break existing implementations which are not intentionally using that as inline breakpoints right ?

**Comment by jonahgraham** [comment]
You may be correct that implementations assume that char start aligns with line start (and char end with line end), I am not sure how to verify that assumption, nor whether it needs consideration.

> IMHO this is a great idea

I stand by this ^^^ part of my comment.
